web-app/uni-todo-app/src/components/TodoListItem.tsx

```diff
@@ -1,10 +1,22 @@
 import { ITodoEntity } from "../store/interfaces/ITodoEntity";
+import { DateTime } from 'luxon';
+
 
 interface ITodoListItemProps {
   todoItem: ITodoEntity;
   onCompleteTodo: () => void;
   onDeleteTodo: () => void;
 }
+const formatDateTime = (date: string): string => {
+  return DateTime.fromISO(date).toLocaleString();
+};
+
+const calculateDaysAgo = (date: string): string => {
+  const updatedAt = DateTime.fromISO(date);
+  const currentDate = DateTime.now();
+  const daysAgo = Math.floor(currentDate.diff(updatedAt, 'days').days);
+return daysAgo.toString();
```

Comment:
indentation looks off here. 
Can you check a verify if a use of `useCallback` or `useMemo` hook would be beneficial here?

Reply:
Thank you for your suggestion!

Indeed is seems that the utilization of useCallback helps in optimizing the performance of the application by ensuring that the functions are not recreated on every render unless their dependencies change. 

By introducing useMemo, we can optimize the computational complexity of the 'daysAgo' calculation. The value of 'daysAgo' will be computed only once and stored, reducing the computational cost of differences from O(n) to O(1), as long as the values used by useMemo (updatedAt) remain constant.
My commit is on the way!
Best regards , Dionisis.